src/NNLSSolverInterface.jl

```diff
@@ -0,0 +1,102 @@
+module NNLSSolverInterface
+
+export QPSolver
+
+using NNLS: QPWorkspace, load!, solve!
+importall MathProgBase.SolverInterface
+
+immutable QPSolver <: AbstractMathProgSolver
+    workspace::QPWorkspace{Float64, Int}
+
+    QPSolver() = new(QPWorkspace{Float64, Int}(0, 0))
+end
+
+type QPModel <: AbstractLinearQuadraticModel
+    workspace::QPWorkspace{Float64, Int}
+    lb::Vector{Float64}
+    ub::Vector{Float64}
+    A::Matrix{Float64}
+    constr_lb::Vector{Float64}
+    constr_ub::Vector{Float64}
+    Q::Matrix{Float64}
+    q::Vector{Float64}
+    sense::Symbol
+    solution::Vector{Float64}
+    duals::Vector{Float64}
+
+    QPModel(work::QPWorkspace) = new(work)
+end
+
+LinearQuadraticModel(s::QPSolver) = QPModel(s.workspace)
+
+numvar(work::QPWorkspace) = size(work.G, 2)
+numconstr(work::QPWorkspace) = size(work.G, 1)
+
+numvar(m::QPModel) = numvar(m.workspace)
+numconstr(m::QPModel) = numconstr(m.workspace)
+
+function loadproblem!(m::QPModel, A, lb, ub, q, constr_lb, constr_ub, sense)
+    m.sense = sense
+    m.A = A
+    m.lb = lb
+    m.ub = ub
+    m.constr_lb = constr_lb
+    m.constr_ub = constr_ub
+    m.q = q
+end
+
+setquadobj!(m::QPModel, Q) = m.Q = Q
+setquadobj!(m::QPModel, rowidx, colidx, quadval) = m.Q = sparse(rowidx, colidx, quadval)
+
+function optimize!(m::QPModel)
+    nvars = size(m.A, 2)
+    nrows = size(m.A, 1)
+    @assert nrows == length(m.constr_lb)
+    @assert nrows == length(m.constr_ub)
+    @assert nvars == length(m.lb)
+    @assert nvars == length(m.ub)
+    @assert nvars == length(m.q)
+
+    nconstr = 2 * nrows + 2 * nvars
+
+    if nvars != numvar(m.workspace) || nconstr != numconstr(m.workspace)
+        resize!(m.workspace, nconstr, nvars)
+        m.solution = fill(NaN, nvars)
+    end
+
+    m.workspace.L = m.Q
```

Comment:
I would do a copy here as well, i.e. `m.workspace.L[:] = m.Q`. Otherwise, what is the `resize!` even for? Also, now there is yet another way for the dimensions to get out of sync.